src/go/k8s/pkg/resources/configmap.go

```diff
@@ -87,6 +87,7 @@ func (r *ConfigMapResource) Ensure(ctx context.Context) error {
 
 // Obj returns resource managed client.Object
 func (r *ConfigMapResource) Obj() (k8sclient.Object, error) {
+	// TODO Retrieve cluster domain dynamically and remove hardcoded cluster.local
 	serviceAddress := r.pandaCluster.Name + "." + r.pandaCluster.Namespace + ".svc.cluster.local"
```

Comment:
also use the new method here

Reply:
I need to ask a follow up question. Do you want to create `svc.DnsName()` in the `ServiceResource` that return only service FQDN? If yes, why there should be coupling between `ConfigMapResource` and `ServiceResource`?

Reply:
I coupled `ConfigMapResource` with `ServiceResource`.